agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/perfcounter/AbstractJmxPerformanceCounter.java

```diff
@@ -24,6 +26,8 @@ public abstract class AbstractJmxPerformanceCounter implements PerformanceCounte
   private final Collection<JmxAttributeData> attributes;
   private boolean alreadyLogged = false;
 
+  private final Set<String> invalidJmxMetrics = new HashSet<>();
```

Comment:
I forget if this class is only used by a single thread

```suggestion
  private final Set<String> invalidJmxMetrics = ConcurrentHashMap.newKeySet();
```

Reply:
It does not seem but I take your suggestion just in case.